src/hdmf/build/manager.py

```diff
@@ -497,12 +497,53 @@ def __ischild(self, dtype):
 
     @staticmethod
     def __set_default_name(docval_args, default_name):
-        new_docval_args = []
-        for x in docval_args:
-            if x['name'] == 'name':
-                x['default'] = default_name
-            new_docval_args.append(x)
-        return new_docval_args
+        if default_name is not None:
+            for x in docval_args:
+                if x['name'] == 'name':
+                    x['default'] = default_name
+
+    def build_docval(self, base, addl_fields, name=None, default_name=None):
```

Comment:
```suggestion
    def __build_docval(self, base, addl_fields, name=None, default_name=None):
```

Reply:
If this is not meant to be called by other classes, make it "private". Otherwise, please add unit tests for this method